### ISSUE TITLE
Unable to select option in make, seeing ^M when hitting enter

### ISSUE THREAD

**Issue by ChrisBAshton** [alphagov/govuk-docker#631]
Running `make support`, I'd get:

```
/Users/christopher.ashton/govuk/govuk-docker/bin/branch_checks.sh support
Fetching recent updates for support
You are not currently on the main branch for support (your branch: dependabot/bundler/redis-5.0.5)
Choose an option
1) Checkout main
2) Ignore
3) Quit
#? 2^M
```

It just wouldn't take my chosen input of '2'.

I tried a simple `select option in "First" "Second" "Third"; do echo "You chose $REPLY"; done` (a simplified derivative of [the code that was failing for me in govuk-docker](https://github.com/alphagov/govuk-docker/blob/043137493f76d82020d3730babd78d508e25276e/bin/branch_checks.sh#L20-L31)) directly in my terminal and that worked, so I thought this was either an issue with the `branch_checks.sh` file, or it not being compatible with my ZSH setup.

Eventually overwrote the entire file with that minimal example, and realised that was failing, i.e. the bash select functionality worked directly on the CLI but always failed when run from a script.

Did a _lot_ of Googling and eventually found the answer here: https://unix.stackexchange.com/a/580621

TLDR: running `stty sane` fixed the issue for me.

Documenting here in case it comes in handy for someone in the future.
